pulp_puppet_plugins/pulp_puppet/plugins/importers/metadata.py

```diff
@@ -169,20 +156,18 @@ def _extract_non_standard_json(filename, temp_dir):
         tgz = tarfile.open(name=filename)
         tgz.extractall(path=extraction_dir)
         tgz.close()
-    except Exception, e:
+    except Exception:
         raise InvalidTarball(filename), None, sys.exc_info()[2]
 
     try:
-        # Recursively look for the metadata file
-        metadata_file_dir = _find_file_in_dir(extraction_dir, constants.MODULE_METADATA_FILENAME)
-
-        if metadata_file_dir is None:
+        # Attempt to find the metadata in the first folder of the tar.gz
+        module_dir = os.listdir(extraction_dir)[0]
```

Comment:
The order won't matter because there should be exactly one folder contained in these `tar.gz` files.

There has to be at least one directory to be a valid Puppet module (files can't just live in the `tar.gz` itself). Pulp expects each individual `tar.gz` to contain exactly one Puppet module. I think Pulp should treat additional directories outside of the Puppet module as an invalid Puppet module.

As an alternative, we could iterate through the top level directories, but I think we should reject them as invalid if this doesn't find metadata in the first arbitrary folder. We do need to document these things which I'm including in the Puppet story along with the API deprecations we talked about, and consolidating the metadata search to not have two implementations.